src/app/components/practice/question-authoring-page/question-authoring.guard.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { TestBed } from '@angular/core/testing';
+
+import { QuestionAuthoringGuard } from './question-authoring.guard';
+import { AuthenticationService } from '../../../services/authentication.service';
+import { User } from '../../../models/user';
+import { UserPermission } from '../../../models/permissions';
+import { of, Observable } from 'rxjs';
+import { Router } from '@angular/router';
+
+describe('QuestionAuthoringGuard', () => {
+  let guard: QuestionAuthoringGuard;
+  let authService: AuthenticationService;
+  const routeStateMock: any = { snapshot: {}, url: ''};
+  const routeMock: any = { snapshot: {}};
+  const routerMock = {navigate: jasmine.createSpy('navigate')};
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [QuestionAuthoringGuard, { provide: Router, useValue: routerMock }, ]
+    });
+    guard = TestBed.inject(QuestionAuthoringGuard);
+    authService = TestBed.inject(AuthenticationService);
+  });
+
+  it('should be created', () => {
+    expect(guard).toBeTruthy();
+  });
+
+
+  it('should allow the authenticated user to access app', () => {
```

Comment:
As this is a unit test, it's probably worth repeating the scenario for the user's we would expect to be able to access this. For example, `org-admin` should be able to create questions and `creator` should be able to do likewise.